### PR TITLE
done1

### PR DESCRIPTION
**Pull request by panianchuk**
https://panianchuk.github.io/react_static-list-of-posts/

### CODE REVIEW

File: src/components/types/Comments.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface Comments {
```
Comment (SanyaBratashchuk) [inline src/components/types/Comments.ts:1]:
```suggestion
export interface Comment {
```

File: src/components/CommentInfo/CommentInfo.tsx [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+        <div className="CommentInfo__title">
+          <strong className="CommentInfo__name">{name}</strong>
+
+          {' by '}
+
+          <a
+            className="CommentInfo__email"
+            href={`mailto:${email}`}
+          >
+            Telly_Lynch@karl.co.uk
```
Comment (SanyaBratashchuk) [inline src/components/CommentInfo/CommentInfo.tsx:27]:
```suggestion
            Telly_Lynch@karl.co.uk
```
Why you hardcode it?

File: src/components/CommentList/CommentList.tsx [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+import { Comments } from '../types/Comments';
+import './CommentList.scss';
 
-export const CommentList: React.FC = () => (
-  <>Put the list here</>
+type Props = {
+  comments: Comments[];
+};
+
+export const CommentList: React.FC<Props> = ({ comments }) => (
```
Comment (SanyaBratashchuk) [inline src/components/CommentList/CommentList.tsx:10]:
```suggestion
export const CommentsList: React.FC<Props> = ({ comments }) => (
```

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,109 +1,30 @@
...
-// import usersFromServer from './api/users';
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+import { PostList } from './components/PostList';
+import { Comment } from './components/types/Comment';
+import { User } from './components/types/User';
+import { Post } from './components/types/Post';
+
+function getUser(userId: number): User | null {
```
Comment (DarkMistyRoom) [inline src/App.tsx:11]:
```suggestion
function getUserById(userId: number): User | null {
```

Comment (DarkMistyRoom) [reply src/App.tsx:11]: image

File: src/App.tsx [hunk 5]
```diff
@@ -1,109 +1,30 @@
...
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function getComments(postId: number): Comment[] {
+  return commentsFromServer.filter(comment => comment.postId === postId);
+}
```
Comment (DarkMistyRoom) [inline src/App.tsx:17]: same here

File: src/components/PostInfo/PostInfo.tsx [hunk 6]
```diff
@@ -1,5 +1,39 @@
...
+        {comments.length === 0 ? (
+          <>
+            <hr />
+            <b data-cy="NoCommentsMessage">No comments yet</b>
+          </>
+        ) : (
+          <CommentsList comments={comments} />
+        )}
+      </div>
+    </>
```
Comment (DarkMistyRoom) [inline src/components/PostInfo/PostInfo.tsx:37]: Don't use React fragment, if you already have a wrapper

File: src/components/CommentList/CommentList.tsx [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+};
+
+export const CommentsList: React.FC<Props> = ({ comments }) => (
+  <>
+    <div className="CommentsList">
+      {comments.map((comment) => (
+        <CommentInfo comment={comment} key={comment.id} />
+      ))}
+    </div>
+  </>
```
Comment (DarkMistyRoom) [inline src/components/CommentList/CommentList.tsx:17]: Don't use React fragment, if you already have a wrapper

Comment (DarkMistyRoom) [reply src/components/CommentList/CommentList.tsx:17]: fix it everywhere

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]
Good job, pls fix comments

**Review by SanyaBratashchuk — APPROVED** [review]

**Pull request closed without merging**